### PR TITLE
Fix bad swizzle on vec4 test outputs

### PR DESCRIPTION
**Pull request by JGamache-autodesk**

### CODE REVIEW

File: source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp [hunk 1]
```diff
@@ -301,9 +301,19 @@ ShaderPtr GlslFragmentGenerator::generate(const string& fragmentName, ElementPtr
             }
             else
             {
-                if (!outputSocket->getType()->isFloat3())
+                if (context.getOptions().hwTransparency)
+                {
```
Comment (JGamache-autodesk) [inline source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp]: To match declared return type from line 182.

Comment (ppenenko) [reply source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp]: Right on, this came up in my HLSL tests as well, and your fix is better: https://github.com/autodesk-forks/MaterialX/pull/694/files/539c06b77a128d04e43ae91312823f55efd7ca9a

File: source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp [hunk 2]
```diff
@@ -314,13 +318,20 @@ ShaderPtr GlslFragmentGenerator::generate(const string& fragmentName, ElementPtr
...
 
-            if (!outputSocket->getType()->isFloat3())
+            if (!context.getOptions().hwTransparency && !outputSocket->getType()->isFloat3())
             {
                 string finalOutput = outputSocket->getVariable() + "_tmp";
                 emitLine(_syntax->getTypeName(outputSocket->getType()) + " " + finalOutput + " = " + outputValue, pixelStage);
                 toVec3(outputSocket->getType(), finalOutput);
                 emitLine("return " + finalOutput, pixelStage);
             }
+            else if (context.getOptions().hwTransparency && !outputSocket->getType()->isFloat4())
```
Comment (ppenenko) [inline source/MaterialXGenOgsXml/GlslFragmentGenerator.cpp]: Could use some refactoring later.

**Comment by ppenenko** [comment]
Travis CI failure looks spurious, I've restarted the job.

**Pull request merged**